homework-04/index.js

```diff
@@ -0,0 +1,111 @@
+const fs = require('fs');
+const path = require('path');
+const { Readable, Transform } = require('stream');
+
+class SentenceStream extends Readable {
+    constructor(sentences, targetSize) {
+        super();
+        this.sentences = sentences;
+        this.targetSize = targetSize;
+        this.size = 0;
+    }
+
+    _read() {
+        if (this.size >= this.targetSize) {
```

Comment:
this can potentially lead to incorrect output file size, since at this point there can be already more bytes in the stream than needed.

to avoid this it's better to check the buffer size before pushing it to stream and cut it if needed.